projects/rocprofiler-systems/source/lib/core/trace_cache/perfetto_processor.cpp

```diff
@@ -0,0 +1,1143 @@
+        [](rocprofiler_memory_allocation_operation_t op) -> const char* {
+        switch(op)
+        {
+            case ROCPROFILER_MEMORY_ALLOCATION_NONE: return "NONE";
+            case ROCPROFILER_MEMORY_ALLOCATION_ALLOCATE: return "ALLOCATE";
+            case ROCPROFILER_MEMORY_ALLOCATION_VMEM_ALLOCATE: return "VMEM_ALLOCATE";
+            case ROCPROFILER_MEMORY_ALLOCATION_FREE: return "FREE";
+            case ROCPROFILER_MEMORY_ALLOCATION_VMEM_FREE: return "VMEM_FREE";
+            default: return "UNKNOWN";
+        }
+    };
+
+    auto _thrd_id    = _mas.thread_id;
+    auto _corr_id    = _mas.correlation_id_internal;
+    auto _stream_id  = _mas.stream_handle;
+    auto _beg_ts     = _mas.start_timestamp;
+    auto _end_ts     = _mas.end_timestamp;
+    auto _addr_val   = _mas.address_value;
+    auto _alloc_size = _mas.allocation_size;
+
+    const auto invalid_context = ROCPROFILER_CONTEXT_NONE;
+    if(_mas.agent_id_handle != invalid_context.handle)
+    {
+        const auto* operation = memop_to_string(
+            static_cast<rocprofiler_memory_allocation_operation_t>(_mas.operation));
+
+        auto _track_desc = [](int32_t _device_id_v, rocprofiler_thread_id_t _tid) {
+            const auto& _tid_v = thread_info::get(_tid, SystemTID);
+            return JOIN("", "GPU Memory Allocation to Agent [", _device_id_v, "] Thread ",
+                        _tid_v->index_data->sequent_value);
+        };
+
+        auto _agent_logical_node_id =
+            m_agent_manager.get_agent_by_handle(_mas.agent_id_handle).logical_node_id;
+
+        const auto _track =
+            tracing::get_perfetto_track(category::rocm_memory_allocate{}, _track_desc,
+                                        _agent_logical_node_id, _thrd_id);
+
+        auto add_perfetto_annotations = [&](::perfetto::EventContext ctx) {
+            if(!m_use_annotations) return;
+
+            annotate_perfetto(ctx, { { "begin_ns", _beg_ts },
+                                     { "end_ns", _end_ts },
+                                     { "corr_id", _corr_id },
+                                     { "stream_id", _stream_id },
+                                     { "bytes", _alloc_size },
+                                     { "agent_id", _agent_logical_node_id },
+                                     { "address", _addr_val } });
+        };
+
+        tracing::push_perfetto(category::rocm_memory_allocate{}, operation, _track,
+                               _beg_ts, ::perfetto::Flow::ProcessScoped(_corr_id),
+                               add_perfetto_annotations);
+        tracing::pop_perfetto(category::rocm_memory_allocate{}, "", _track, _end_ts);
+    }
+#endif
+}
+
+void
+perfetto_processor_t::handle([[maybe_unused]] const region_sample& _rs)
+{
+    auto _corr_id  = _rs.correlation_id_internal;
+    auto _beg_ts   = _rs.start_timestamp;
+    auto _end_ts   = _rs.end_timestamp;
+    auto _category = _rs.category;
+    auto _name     = _rs.name;
```

Comment:
I think all these values can be declared as a const

```suggestion
    auto _corr_id  = _rs.correlation_id_internal;
    auto _beg_ts   = _rs.start_timestamp;
    auto _end_ts   = _rs.end_timestamp;
    auto _category = _rs.category;
    auto _name     = _rs.name;
```